hoomd/md/ForceComposite.cc

```diff
@@ -630,6 +630,11 @@ CommFlags ForceComposite::getRequestedCommFlags(uint64_t timestep)
     // request orientations
     flags[comm_flag::orientation] = 1;
 
+    // request velocity, angular momentum, and inertia
+    flags[comm_flag::velocity] = 1;
+    flags[comm_flag::angmom] = 1;
+    flags[comm_flag::inertia] = 1;
```

Comment:
I think these requests should be in PotentialPairDPDThermo. That is currently the only place that uses `comm_flag::velocity` that would be compatible with rigid bodies.

`ForceDistanceConstraint` also sets the velocity flag, but that isn't compatible with rigid bodies.

Reply:
If these flags are not set by `ForceComposite`, `updateRigidBodies` will need to check if this information is being communicated to decide if it can set the constituent velocities. This will lead to diverging behavior in serial and MPI simulations, since in serial `updateRigidBodies` will always set the constituent velocities correctly, while in MPI simulations it will depend on which other objects have been attached to the communicator.

If we wanted to implement this behavior, I would suggest a subscriber model for `ForceComposite` to request the constituent velocities be updated, then `ForceComposite` would only request this information if that was true. This seems like more work than it's worth, though.

Reply:
Understood.